src/notebooks/controllers/vscodeNotebookController.ts

```diff
@@ -211,6 +212,16 @@ export class VSCodeNotebookController implements Disposable, IVSCodeNotebookCont
         traceInfoIfCI(`${ConsoleForegroundColors.Green}Posting message to Notebook UI ${messageType}`);
         return this.controller.postMessage(message, editor);
     }
+    /**
+     * A cell has been added to the notebook, so wait for the execution to be queued before handling any more execution requests.
+     * This only applies to the Interactive Window since cells are added from both the extension and core.
+     * @param promise A promise that resolves when the notebook is ready to handle more executions.
+     */
+    public setPendingCellAddition(promise: Promise<void>): void {
```

Comment:
Feels a bit funny to have this just hanging off the main controller when it only applies to the IW. No reason other users couldn't just call it and it does nothing without a warning or error. 

Could we consider something like extending IVsCodeNotebookController to IVsCodeInteractiveNotebookController with viewType locked to InteractiveWindowView and this function available? Or maybe an error or warning if you call it from a notebook?

Reply:
If we were to change this, I'd suggest throwing an error instead of a warning, as it should never be called for non-IW

Reply:
an IW override does sound like a clearer solution, but I'd probably still just throw for the notebook implementation to avoid a cast. I'll stick with a type check for now to avoid having to deal with the controller registration changes that would be necessary for the polymorphism solution.